unitTests/ui/TemplateFieldsEvaluatorTest.ts

```diff
@@ -18,41 +20,72 @@ export function TemplateFieldsEvaluatorTest() {
     });
 
     it('should be able to evaluate fields to match', () => {
-      fieldsToMatch = [
-        {
-          field: 'foo',
-          values: ['bar', 'baz']
-        }
-      ];
+      fieldsToMatch = [{ field, values: [fieldValue, 'baz'] }];
       expect(TemplateFieldsEvaluator.evaluateFieldsToMatch(fieldsToMatch, result)).toBe(true);
     });
 
     it('should be able to evaluate fields that do not match', () => {
-      fieldsToMatch = [
-        {
-          field: 'foo',
-          values: ['brak']
-        }
-      ];
+      fieldsToMatch = [{ field: 'foo', values: ['brak'] }];
       expect(TemplateFieldsEvaluator.evaluateFieldsToMatch(fieldsToMatch, result)).toBe(false);
     });
 
     it('should be able to evaluate fields that are not null', () => {
-      fieldsToMatch = [
-        {
-          field: 'foo'
-        }
-      ];
+      fieldsToMatch = [{ field }];
       expect(TemplateFieldsEvaluator.evaluateFieldsToMatch(fieldsToMatch, result)).toBe(true);
     });
 
     it('should be able to evaluate fields that are null', () => {
-      fieldsToMatch = [
-        {
-          field: 'bar'
-        }
-      ];
+      fieldsToMatch = [{ field: 'bar' }];
       expect(TemplateFieldsEvaluator.evaluateFieldsToMatch(fieldsToMatch, result)).toBe(false);
     });
+
+    it(`when the fieldsToMatch #values is an array containing the field value as uppercase,
+    it returns true`, () => {
+      fieldsToMatch = [{ field, values: [fieldValue.toUpperCase()] }];
+      expect(TemplateFieldsEvaluator.evaluateFieldsToMatch(fieldsToMatch, result)).toBe(true);
+    });
+
+    it(`given the result raw field has an uppercase value,
+    when the fieldsToMatch #values is an array containing the field value as lowercase,
+    it returns true`, () => {
+      result.raw[field] = fieldValue.toUpperCase();
+      fieldsToMatch = [{ field, values: [fieldValue.toLowerCase()] }];
+
+      expect(TemplateFieldsEvaluator.evaluateFieldsToMatch(fieldsToMatch, result)).toBe(true);
+    });
+
+    describe(`when the result raw has a field that has an array of values (e.g. a multi-value field)`, () => {
+      const multiValueField = 'language';
+      const multiValueFieldValues = ['francais'];
```

Comment:
I think we are missing some test where the `multiValueFieldValues` contains multiple value : 

* `multiValueFieldValues = ['francais', 'spanish', 'english']` --> try match on one or multiple of them, or only some of them etc..